setup.py

```diff
@@ -40,6 +50,8 @@ def main():
                   'Oliver Fricko, Peter Kolp',
         "author_email": 'message_ix@iiasa.ac.at',
         "url": 'http://github.com/iiasa/message_ix',
+        "install_requires": INSTALL_REQUIRES,
```

Comment:
@khaeru I am not so familiar with setting up python packages. Could you explain a bit how does `install_requires` and `extra_require` work? Is it now not necessary to run `pip install -r requirements.txt` - so `python setup.py install` will do the magic?

Reply:
`pip install .` should be the only installation command. This will:
1. Download and install all the packages from `install_requires`, if the user doesn't already have them.
2. Implicitly call `setup.py install` to install the package in the current directory (".").

Calling (for instance) `pip install .[docs]` means "install the current package, and also the 'docs' extra features." So the dependencies specified in `extras_require['docs']` will also be downloaded and installed. Multiple extras can be specified: `pip install .[docs,tests]`.

**More details:** Using a requirements.txt is okay for CI tools, where one can give a script that does many things as part of the installation process.

But an end-user installing the package from PyPI will run `pip install ixmp`. In this case, pip will only understand dependencies given in setup.py.

I think this issue hasn't come up before now because Anaconda was preferred for ixmp/message_ix. Anaconda packaging has an entirely different format/workflow, so [the dependencies are specified in a different place](https://github.com/gidden/ixmp-feedstock/blob/master/recipe/meta.yaml#L24) and it didn't make a difference that they were missing from setup.py.

Reply:
Thanks for explanation. I added `nbformat` and `jupyter` as extra dependencies for tests to `setup.py` (see last commit) to make tests working. Should we update readme to replace `pip install -r requirements.txt` with `pip install .`?

Reply:
Just pushed a commit updating the README.

I think that `nbformat` is a dependency of `nbconvert` which is in turn a dependency of `jupyter`. So probably just `jupyter` would be sufficient; did you try that?

Reply:
Yes, just tried and it worked. Thanks!